src/core/utilities/random.ts

```diff
@@ -19,3 +22,23 @@ export const getRandomString = (length?: number): string => {
 
   return string;
 };
+
+/**
+ * getRandomHex() returns a randomly generated hexadecimal string of the given
+ * length. If no length is provided, the result will be 32 characters. Note
+ * that this function does not use cryptographically secure randomness and as
+ * such should not be used for cryptographic purposes.
+ *
+ * @param length - Optional. The length of the string to generate
+ * @returns A randomly generated hexadecimal string
+ */
+export const getRandomHex = (length?: number): HexString => {
+  let string = "";
+
+  for (let i = 0; i < (length || 32); i++) {
+    const index = Math.floor(Math.random() * HEX.length);
+    string += HEX[index];
+  }
+
+  return `0x${string}`;
```

Comment:
Side note: We need to be better about what requires and what doesn't require a `0x` prefix....